File: src/main/java/de/php_perfect/intellij/ddev/actions/DdevDeleteAction.java
package de.php_perfect.intellij.ddev.actions;

import com.intellij.openapi.project.Project;
import de.php_perfect.intellij.ddev.state.DdevStateManager;
import de.php_perfect.intellij.ddev.state.State;
import org.jetbrains.annotations.NotNull;

public final class DdevDeleteAction extends DdevRunAction {
    public DdevDeleteAction() {
        super("delete");
    }

    protected boolean isActive(@NotNull Project project) {
        final State state = DdevStateManager.getInstance(project).getState();

        return state.isInstalled() && state.isConfigured();
    }
}

File: src/main/java/de/php_perfect/intellij/ddev/actions/DdevDeleteAction.kt
package de.php_perfect.intellij.ddev.actions

import com.intellij.openapi.project.Project
import de.php_perfect.intellij.ddev.cmd.DdevRunner
import de.php_perfect.intellij.ddev.state.DdevStateManager
import de.php_perfect.intellij.ddev.state.State
import org.jetbrains.annotations.NotNull

class DdevDeleteAction : DdevRunAction() {
    override fun run(@NotNull project: Project) {
        DdevRunner.getInstance().delete(project)
    }

    override fun isActive(@NotNull project: Project): Boolean {
        val state: State = DdevStateManager.getInstance(project).state
        return state.isAvailable && state.isConfigured
    }
}
